crates/broker/src/order_monitor.rs

```diff
@@ -680,18 +708,35 @@ where
                         proof_time_seconds,
                         completion_time
                     );
-                    self.db
-                        .set_order_status(&order.id(), OrderStatus::Skipped)
-                        .await
-                        .context("Failed to set order status to skipped")?;
+                    // Set order to skipped in DB and continue.
+                    self.skip_order(&order, "peak KHz limit: completion past expiration").await;
```

Comment:
Not for this PR, but this is something I've been thinking about. The fact that we set it to skipped here means it is not possible to pick this order up if capacity becomes available. During a load spike, this means orders will be dropped if all provers are over-committed vs slowing down and having prices rise. It would be good to keep these order around, and reassess on each iteration whether we have capacity to take it on. In order to make this possible, we need to make sure that iterating over the orders and checking it against capacity is cheap to do, which it is not obviously so here (also not obviously too expensive, I just haven't really looked close enough).

Reply:
ah yeah true, I should just remove the skip here, but not pass the order along. I thought I was doing that, just forgot to update this

Reply:
https://github.com/boundless-xyz/boundless/pull/668/commits/0173a2fa4853178e7c174f5a3dbd6e1cb690dc97

Not sure why I kept this behaviour around from previous logic

Reply:
https://github.com/boundless-xyz/boundless/pull/668/commits/a0c5450eb05b05c2756f5690f604cafb0cc3e394

Decided to keep the skip in the case where the broker cannot fulfill even if started immediately and prioritized. Seems like the best to not keep around orders that are unreasonable until expiry.